.buildkite/filebeat/filebeat-pipeline.yml

```diff
@@ -1,5 +1,157 @@
+    steps:
+      - label: ":ubuntu: Unit Tests"
+        command:
+          - "sudo apt-get update"
+          - "sudo apt-get install -y libsystemd-dev"
+          - "sudo apt install -y python3-pip"
+          - "sudo apt-get install -y python3-venv"
+          - ".buildkite/filebeat/scripts/unit-tests.sh"
+        notify:
+          - github_commit_status:
+              context: "Filebeat/Mandatory: Unit Tests"
+        agents:
+          provider: "gcp"
+          image: "${IMAGE_UBUNTU_X86_64}"
+
+      - label: ":ubuntu: Integration Tests"
+        command:
+          - "sudo apt-get update"
+          - "sudo apt-get install -y libsystemd-dev"
+          - ".buildkite/filebeat/scripts/integration-tests.sh"
+        notify:
+          - github_commit_status:
+              context: "Filebeat/Mandatory: Integration Tests"
+        agents:
+          provider: "gcp"
+          image: "${IMAGE_UBUNTU_X86_64}"
+
+      - label: ":ubuntu: Python Integration Tests"
+        command:
+          - "sudo apt-get update"
+          - "sudo apt-get install -y libsystemd-dev"
+          - ".buildkite/filebeat/scripts/integration-pytests.sh"
+        notify:
+          - github_commit_status:
+              context: "Filebeat/Mandatory: Python Integration Tests"
+        agents:
+          provider: "gcp"
+          image: "${IMAGE_UBUNTU_X86_64}"
+
+      - label: ":windows:-{{matrix.image}} Unit Tests"
+        command: ".buildkite/filebeat/scripts/unit-tests-win.ps1"
+        notify:
+          - github_commit_status:
+              context: "Filebeat/Mandatory: {{matrix.image}} Unit Tests"
+        agents:
+          provider: "gcp"
+          image: "{{matrix.image}}"
+          machine_type: "n2-standard-8"
+          disk_size: 200
+          disk_type: "pd-ssd"
+        matrix:
+          setup:
+            image:
+              - "${IMAGE_WIN_2016}"
+              - "${IMAGE_WIN_2022}"
+
+  - group: "Extended Testing"
+    key: "extended-tests"
+    if: build.env("BUILDKITE_PULL_REQUEST") != "false"
+
+    steps:
+      - label: ":linux: ARM64 Unit Tests"
+        key: "arm-extended"
+        if: build.env("GITHUB_PR_TRIGGER_COMMENT") == "filebeat for arm" || build.env("GITHUB_PR_LABELS") =~ /.*arm.*/
+        command:
+          - "sudo apt-get update"
+          - "sudo apt-get install -y libsystemd-dev"
+          - "sudo apt install -y python3-pip"
+          - "sudo apt-get install -y python3-venv"
+          - ".buildkite/filebeat/scripts/unit-tests.sh"
+        notify:
+          - github_commit_status:
+              context: "Filebeat/Extended: Unit Tests ARM"
+        agents:
+          provider: "aws"
+          imagePrefix: "${IMAGE_UBUNTU_ARM_64}"
+          instanceType: "t4g.large"
+
+      - label: ":mac: MacOS Unit Tests"
+        key: "macos-extended"
+        if: build.env("GITHUB_PR_TRIGGER_COMMENT") == "filebeat for macos" || build.env("GITHUB_PR_LABELS") =~ /.*macOS.*/
+        command:
+          - "ulimit -Sn 50000"
```

Comment:
If possible, I'd say this should be included in a specific script with the rationale for using this. What do you think?